src/newtype.rs

```diff
@@ -0,0 +1,57 @@
+use crate::constants::*;
+use rusqlite as sql;
+use rusqlite::types::*;
+use std::fmt;
+
+macro_rules! defnewtype {
+    ($name:ident ($base:ty); $( $field:ident => $value:expr),*) => {
+        #[repr(transparent)]
+        #[derive(Copy, Clone, Eq, PartialEq, Debug)]
+        pub struct $name { value: $base }
```

Comment:
this should be an enum, not a struct unless I am missing sth major

Reply:
Your suggestion to use `enum` and `num::FromPrimitive` sounds reasonable, but `ToPrimitive` does not quite fit. Type of `to_i64' is `Self -> Optiion<i64>`, which is too wide.

On why I decided to use `struct`, not `enum`. You can have `enum` with `#[repr(i32)]` or `#[repr(u32)]`, but not `#[repr(libc::c_int)]`. There is no standard that warrants that `i32` and `u32` have same representation, there is even nothing that promises `sizeof(int) == 4`.

As such, using `enum` of `struct` would result in:

 * number of `to_i64().unwrap()` here and there, which looks creepy.
 * potential ABI breakage on some platforms. 

If you are sure this is okay, I will switch to `enum`, no problem.

Reply:
On second thought, `dc_msg_t` is abstract data type, so I was wrong, ABI problems do not arise. So, `enum` is probably fine. Moment.